create.cjs

```diff
@@ -701,3 +681,88 @@ function generateDonwloadStats() {
     ).join('\n')
   );
 }
+
+const ignoredFilesInPackages = new Set(['dist', 'node_modules', 'package.json']);
+/**
+ * @param {VirtualPackage} pkg
+ */
+async function writePackage(pkg) {
+  await fsPromises.mkdir(pkg.path, { recursive: true });
+  let hasChanged = false;
+
+  const promises = [];
+
+  /** @type {Set<string>} */
+  const existingFileFullpaths = new Set();
+
+  const ps = new PathScurry(pkg.path);
+  for await (const file of ps) {
+    if (file.name[0] === '.' || ignoredFilesInPackages.has(file.name)) {
+      continue;
+    }
+
+    if (file.isFile()) {
+      const relativePath = file.relativePosix();
+      if (!(relativePath in pkg.files)) {
+        // remove extra files
+        hasChanged = true;
+
+        promises.push(
+          fsPromises.rm(path.join(pkg.path, relativePath))
+        );
+      } else {
+        existingFileFullpaths.add(file.fullpathPosix());
+      }
+    }
+  }
+
+  const packageJsonPath = path.join(pkg.path, 'package.json');
+
+  // write files, and check if they changed
+  Object.entries(pkg.files).forEach(([file, content]) => {
+    const filePath = path.join(pkg.path, file);
+    promises.push(
+      compareAndWriteFile(filePath, content, existingFileFullpaths)
+        .then(written => {
+          if (written) {
+            hasChanged = true;
+          }
+        })
+    );
+  });
+
+  // check if package.json changed
+  promises.push((async () => {
+    const existingPackageJson = (
+      existingFileFullpaths.has(packageJsonPath)
+      || await fileExists(packageJsonPath)
+    ) ? JSON.parse(await fsPromises.readFile(packageJsonPath, 'utf8'))
+      : {};
+
+    // exclude version from comparison
+    if (JSON.stringify({ ...existingPackageJson, version: undefined }) !== JSON.stringify({ ...pkg.packageJson, version: undefined })) {
```

Comment:
Don't know which is faster, [`dequal`](https://www.npmjs.com/package/dequal) or `JSON.stringify`. I will create a naive benchmark to test it.

Reply:
`dequal` is 2x faster: https://replit.com/@isukkaw/dequal-vs-json-stringify

So in [`290a697` (#23)](https://github.com/SukkaW/nolyfill/pull/23/commits/290a697268b0e368e13996c8f70cdbfafb4f66f7) I have replaced `JSON.stringify` with `dequal`.